packages/babel-parser/src/parser/expression.js

```diff
@@ -1208,12 +1232,52 @@ export default class ExpressionParser extends LValParser {
           }
         }
       }
+
       // fall through
       default:
         throw this.unexpected();
     }
   }
 
+  // https://github.com/js-choi/proposal-hack-pipes
+  parseTopicReference() {
+    const pipeProposal = this.getPluginOption("pipelineOperator", "proposal");
+
+    if (pipeProposal) {
+      const node = this.startNode();
+
+      const proposalToNodeType = {
+        hack: "TopicReference",
+        smart: "PipelinePrimaryTopicReference",
+      };
+
+      const throwPipeTopicRequiresHackPipesError = () => {
+        throw this.raise(node.start, Errors.PipeTopicRequiresHackPipes);
+      };
+
+      const nodeType =
+        proposalToNodeType[pipeProposal] ??
+        throwPipeTopicRequiresHackPipesError();
```

Comment:
Nit (sorry, I'm commenting on the code style):
1. I don't like using `proposalToNodeType[pipeProposal]` much since it can defer to prototype access
2. This arrow-function-just-to-have-throw-expressoin is clever but I don't think I have ever seen it used in our code base

Could you use something like this instead? :pray:
```suggestion
      let nodeType;
      if (pipeProposal === "hack") nodeType = "TopicReference";
      else if (pipeProposal === "smart") nodeType = "PipelinePrimaryTopicReference";
      else throw this.raise(node.start, Errors.PipeTopicRequiresHackPipes);
```